pkg/scheduler/queue/mapping.go

```diff
@@ -0,0 +1,111 @@
+package queue
+
+var empty = string([]byte{byte(0)})
+
+// Mapping is a map-like data structure that allows accessing its items not
+// only by key but also by index.
+// When an item is removed, the iinternal key array is not resized, but the
+// removed place is marked as empty. This allows to remove keys without
+// changing the index of the remaining items after the removed key.
+// Mapping uses *tenantQueue as concrete value and keys of type string.
+// The data structure is not thread-safe.
+// TODO(chaudum): Implement QueueMapping using generics
```

Comment:
nit: Better to add `O(n)` notation for inserts, scans and deletes for this data structure?. My feeling is that would communicate the efficiency of this datastructure bit better without much words?

Reply:
| Operation | Function | Complexity |
| --------- | -------- | ---------- |
| Insert | `Put(key string, value *tenantQueue) bool` | `O(1)`|
| Select | `GetNext(idx QueueIndex) *tenantQueue` | `O(1)` (best case) `O(n)` (worst case) |
| Remove | `Remove(key string) bool` | `O(1)` |